### PR TITLE
admin: fix `admin recover index` with CommonHandle table in TiKV env | tidb-test=pr/2231 

### LINKED ISSUE #47687: `admin recover index` report error with real tikv
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
create table t(a int, b char(10) collate utf8mb4_bin, c char(10) collate utf8mb4_general_ci,d varchar(10) collate utf8mb4_bin, e varchar(10) collate utf8mb4_general_ci, f char(10) collate utf8mb4_unicode_ci, g varchar(10) collate utf8mb4_unicode_ci, primary key(a, b, c, d, e, f, g), key a(a), unique key ua(a), key b(b), unique key ub(b), key c(c), unique key uc(c),key d(d), unique key ud(d),key e(e), unique key ue(e), key f(f), key g(g), unique key uf(f), unique key ug(g));
insert into t values (1, '啊  ', '啊  ', '啊  ', '啊  ', '啊  ', '啊  ');
admin recover index t a;
```

### 2. What did you expect to see? (Required)

```
run without error
```

### 3. What did you see instead (Required)

```
mysql> admin recover index t a;
ERROR 1105 (HY000): [components/tidb_query_executors/src/table_scan_executor.rs:425]: Data is corrupted, missing data for NOT NULL column (offset = 0)
```

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #47687

Problem Summary: In TiKV env, tableFullScan can't process schema have same columns which make it paniced.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: pkg/executor/test/admintest/admin_test.go [hunk 1]
```diff
@@ -376,6 +376,22 @@ func TestClusteredIndexAdminRecoverIndex(t *testing.T) {
 	tk.MustQuery("admin recover index t idx").Check(testkit.Rows("1 3"))
 	tk.MustQuery("SELECT COUNT(*) FROM t USE INDEX(idx)").Check(testkit.Rows("3"))
 	tk.MustExec("admin check table t;")
+
```
Comment (zimulala) [inline pkg/executor/test/admintest/admin_test.go:379]: Do we need to add a simple comment for the new test case here?

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**